services/processor/native/call.go

```diff
@@ -98,22 +105,38 @@ func (s *service) prepareMethodInputArgsForCall(methodInstance types.MethodInsta
 			}
 			res = append(res, reflect.ValueOf(arg.StringValue()))
 		case reflect.Slice:
-			if methodType.In(i).Elem().Kind() != reflect.Uint8 {
-				return nil, errors.Errorf("method '%s' arg %d slice type is not byte", functionNameForErrors, i)
-			}
-			if !arg.IsTypeBytesValue() {
-				return nil, errors.Errorf("method '%s' expects arg %d to be bytes but it has %s", functionNameForErrors, i, arg.StringType())
+			switch methodTypeIn.Elem().Kind() {
+			case reflect.Uint8:
+				res = append(res, reflect.ValueOf(arg.BytesValue()))
+			case reflect.String:
+				res = append(res, reflect.ValueOf(arg.StringStringValue()))
+			case reflect.Uint32:
+				res = append(res, reflect.ValueOf(arg.Uint32Value()))
+			case reflect.Uint64:
+				res = append(res, reflect.ValueOf(arg.Uint64Value()))
+			case reflect.Slice:
+				if !arg.IsTypeBytesValue() {
+					return nil, errors.Errorf("method '%s' expects arg %d to be bytes but it has %s", functionNameForErrors, i, arg.StringType())
+				}
+				res = append(res, reflect.ValueOf(arg.BytesValue()))
+			default:
+				return nil, errors.Errorf("method '%s' expect arg %d to be of different type", functionNameForErrors, i)
```

Comment:
how does this affect gamma-cli ? there was a limit to number of args there ...

Reply:
I does not affect gamma-cli per se, support for this should be added to gamma-cli.

The only way to make it work is to construct a special transaction and gamma-cli does not know how to do it yet.